src/content/docs/en/guides/incremental-static-regeneration.mdx

```diff
@@ -0,0 +1,218 @@
+
+```ts title="src/middleware.ts"
+
+const cache = await caches.open('astro')
+
+export const onRequest = async ({ request }, next) => {
+    const cachedResponse = await cache.match(request)
+    
+    if (cachedResponse) {
+        const expires = Number(cachedResponse.headers.get('X-expires'))
+        if (expires > Date.now()) {
+            // return cached response
+            return cachedResponse
+        }
+        else {
+            // remove stale response
+            await cache.delete(request)
+        }
+    }
+    
+    const response = await next()
+    
+    // set expiry
+    response.headers.set('X-expires', String(Date.now() + 60 * 1000))
+
+    await cache.put(request.url, response.clone())
+    // return fresh response
+    return response
+}
+```
+
+### Deno Deploy
+Deno Deploy does not provide a way to persist data. However, you may be able to use a third-party caching database to manually cache responses. While this avoids calls to your CMS, you may still experience significant latencies due to the network round-trip time between the Deno Deploy instance and your database.
+
+### Vercel
+Vercel provides a built-in caching layer that automatically saves cacheable responses. Make sure your server-rendered page or endpoint sets a `Cache-Control` header.
+
+```astro title="src/pages/index.astro"
+---
+Astro.response.headers.set('Cache-Control', 's-maxage=60')
+---
+<h1>Welcome to Astro</h1>
+```
+
+```ts title="src/pages/api.ts"
+export async function get({ params, request }) {
+    ...
+    return new Response(content, {
+        headers: { "Cache-Control": "s-maxage=60" }
+    })
+}
+```
+Vercel only supports `s-maxage` and `stale-while-revalidate` in the Cache-Control header. See [vercel documentation](https://vercel.com/docs/concepts/edge-network/caching) for current information.
+
+### Netlify
+Netlify supports caching via two methods: `Cache-Control` headers, which works with edge and serverless functions; and, On-demand Builders, a type of function dedicated for optimized caching.
+
+#### On-demand Builders
+On-demand Builders are serverless functions used to generate web content as needed that’s automatically cached on Netlify’s Edge CDN. They enable you to build 
+To enable them, set the `builders` option to `true` in your config file.
+```ts title="stro.config.js" ins={6}
+import { defineConfig } from 'astro/config'
+import netlify from '@astrojs/netlify/functions'
+
+export default defineConfig({
+  output: 'server',
+  adapter: netlify({ builders: true })
+})
+```
+By default, pages for your site will be built when a user visits them for the first time and then cached at the edge for subsequent visits. To set a revalidation time, call the `runtime.setBuildersTtl(ttl)` local with the duration (in seconds). For example, to set a revalidation time of 60 seconds:
+```ts title="src/pages/index.astro"

```

Comment:
```suggestion
```astro title="src/pages/index.astro"

```

Reply:
Oh, thanks for the catch! I was looking at the wrong code block.